envi/archs/h8/__init__.py

```diff
@@ -43,91 +43,91 @@ def archParseOpcode(self, bytes, offset=0, va=0):
         return self._arch_dis.disasm(bytes, offset, va)
 
     def getEmulator(self):
-        return H8Emulator()
+        import envi.archs.h8.emu as h8_emu
+        return h8_emu.H8Emulator()
 
-from envi.archs.h8.emu import *
 
 memreg_h8s = {  # from rej06b0824_h8sap.pdf (h8s/2400).  uncommented entries from IDA H9S/Advanced
-        0xffff84 : 'SBYCR',     # Standby Control Register
-        0xffff86 : 'MSTPCRH',
-        0xffff87 : 'MSTPCRL',
-        0xffff89 : 'BRR_1',
-        0xffff8a : 'SCR_1',
-        0xffff8b : 'TDR_1',
-        0xffffa0 : 'DADRAH',
-        0xffffa1 : 'BRR_2',
-        0xffffa2 : 'SCR_2',
-        0xffffa3 : 'TDR_2',
-        0xffffa4 : 'SSR_2',
-        0xffffa5 : 'RDR_2',
-        0xffffa8 : 'WDT_0',
-        0xffffa8 : 'TCNT_0',    # Timer Counter_0(write)  - 16bits
-        0xffffa9 : 'TCNT_0_byte1',
-        0xffffaa : 'PAODR',     # Port A Output Data Register
-        0xffffab : 'PA0DDR',    # Port A Data Direction Register
-        0xffffb5 : 'P4ODDR',    # Port 4 Data Direction Register
-        0xffffb7 : 'P4ODR',     # Port 4 Data Register
-        0xffffbb : 'P6DR',
-        0xffffbe : 'P8DDR',
-        0xffffbf : 'P8DR',
-        0xffffc0 : 'P9DDR',
-        0xffffc1 : 'P9DR',
-        0xffffc2 : 'IER',
-        0xffffc3 : 'STCR',
-        0xffffc4 : 'SYSCR',     # System Control Register
-        0xffffc5 : 'MDCR',      # Mode Control Register
-        0xffffc6 : 'BCR',
-        0xffffc7 : 'WSCR',
-        0xffffd8 : 'SMR',
-        0xffffd9 : 'BRR',
-        0xffffda : 'SCR',
-        0xffffdb : 'TDR',
-        0xffffdc : 'SSR',
-        0xffffdd : 'RDR',
+        0xffff84: 'SBYCR',     # Standby Control Register
+        0xffff86: 'MSTPCRH',
+        0xffff87: 'MSTPCRL',
+        0xffff89: 'BRR_1',
+        0xffff8a: 'SCR_1',
+        0xffff8b: 'TDR_1',
+        0xffffa0: 'DADRAH',
+        0xffffa1: 'BRR_2',
+        0xffffa2: 'SCR_2',
+        0xffffa3: 'TDR_2',
+        0xffffa4: 'SSR_2',
+        0xffffa5: 'RDR_2',
+        # 0xffffa8: 'WDT_0',   # XXX: Reaffirm this
```

Comment:
Okay. I marked that down because two dictionary keys pointing at different things means one fo them is getting overrided. I found the data sheet that we reference. Turns out there's a lot of different h8s chips that we can reference, all of which have different mem -> reg mappings. 

Might want to expand on that in the future?

But for now, I guess remove that entry since it's going to get overridden in the first place?

Reply:
makes sense.  there is an overlap in the datasheet, but at the moment i'm not sure the best way to reconcile them.  removing WDT_0 makes the most sense right now.

Reply:
:+1: